### PR TITLE
Move `consider-using-namedtuple-or-dataclass` to `CodeStyle` extension

### LINKED ISSUE #4557: Move `consider-using-namedtuple` to optional extension
### Is your feature request related to a problem? Please describe

Move `consider-using-namedtuple` checker to an optional extension. It shouldn't be activated by default.

### Additional context

Related to #4538

### PR DESCRIPTION
**Pull request by cdce8p**
## Description
Move `consider-using-namedtuple-or-dataclass` checker added in #4517 to new `CodeStyle` extension.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Related Issue
Closes #4557

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -124,9 +124,6 @@ modules are added.
   Closes #4509
   Closes PyCQA/astroid#999
 
-* New checker ``consider-using-namedtuple-or-dataclass``. Emitted when dictionary values
```
Comment (Pierre-Sassoulas) [inline ChangeLog:127]: Don't we still need to advertise this new checker (as optional that need to be added to the pylintrc to be used) ?

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I think this checker is nice and needs to be shown in the changelog and what's new so it's used. I also think this is especially important for an optional one, so power-user that read the changelog knows what do to. (I don't think the ``--all`` option is loading optional checker, but there should be a way to add optional checker to the pylintrc, I need to open this issue about this, but this is a really busy week 😅 .)

File: ChangeLog (hunk 1 above)
Comment (cdce8p) [reply ChangeLog:127]: I already added a note about the `CodeStyleChecker` with one of the last MRs. However I agree, it does make sense to mention the checker as well. I've moved it so it's a sub-point now.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**